src/cmd/flux-account-fetch-job-records.py

```diff
@@ -90,6 +90,24 @@ def fetch_new_jobs(last_timestamp=0.0):
         if data["jobspec"] is not None:
             single_record["jobspec"] = data["jobspec"]
 
+        required_keys = [
+            "userid",
+            "t_submit",
+            "t_run",
+            "t_inactive",
+            "ranks",
+            "id",
+            "R",
+            "jobspec",
+        ]
+        if not all(
+            key in single_record and single_record[key] is not None
```

Comment:
Just a suggestion, but I think you check if `key in single_record and single_record[key] is not None` in one call with `single_record.get(key)` (returns falsy if key doesn't exist or is `None`)

Reply:
I hope I'm not misunderstanding you but I think the check would have to be `single_record.get(key) is not None`: https://docs.python.org/3/library/stdtypes.html#dict.get

Reply:
Oh wait, maybe you're right, assuming that all the values are truthy if they aren't `None`...

Reply:
> I hope I'm not misunderstanding you but I think the check would have to be single_record.get(key) is not None: https://docs.python.org/3/library/stdtypes.html#dict.get

Very good point @jameshcorbett - I should keep my mouth shut about Python stuff.

Reply:
booo everyone is wrong sometimes